fetcher/src/job/job.processor.ts

```diff
@@ -18,17 +22,36 @@ export class JobProcessor extends WorkerHost {
     if (keys.length == 0 || keys.length > 1) {
       throw new FetcherError(FetcherErrorCode.UnexpectedNumberOfJobs, String(keys.length))
     } else {
-      const aggregatorHash = keys[0]
-      const aggregatorId = inData[aggregatorHash].aggregatorId
-      const feeds = inData[aggregatorHash].feeds
+      const adapterHash = keys[0]
+      const aggregatorId = inData[adapterHash].aggregatorId
+      const feeds = inData[adapterHash].feeds
       const data = await fetchData(feeds, this.logger)
       const aggregate = aggregateData(data)
+      const threshold = inData[adapterHash].threshold
+      const oracleAddress = inData[adapterHash].address
+      const aggregatorHash = inData[adapterHash].aggregatorHash
 
       try {
+        const { value: lastSubmission } = await fetchDataFeed({
+          aggregatorHash,
+          logger: this.logger
+        })
         let response = await insertMultipleData({ aggregatorId, timestamp, data })
         this.logger.debug(response)
-        response = await insertAggregateData({ aggregatorId, timestamp, value: aggregate })
+        response = await insertAggregateData({
+          aggregatorId,
+          timestamp,
+          value: aggregate
+        })
         this.logger.debug(response)
+        if (abs(aggregate - Number(lastSubmission)) >= lastSubmission * threshold) {
```

Comment:
This condition is actually not correct because `lastSubmission` and `submission` operate with an extra decimals that are part of adapter definition (e.g. https://config.orakl.network/adapter/ada-usdt.adapter.json)

How about using this already implemented function instead? 

https://github.com/Bisonai/orakl/blob/f4d2b8cdbe8d1d02165471044369f7027e9815ff/fetcher/src/job/job.utils.ts#L119-L153